### PR TITLE
CCITCARBON-291 Fix the show command --vars-data

### PR DESCRIPTION
**Pull request by JunqiZhang0**
Added --vars-data option for show command

### CODE REVIEW

**Review by rujutashinde — COMMENTED** [review]
@JunqiZhang0 i saw in the ticket you have mentioned :+1: Two things to do:

Fix the erorr throwed for sub scenarios in include session with variable not in place
Add vars-data support

I was not sure what you meant by fix the error thrown in include section, is there something more to be added in this fix?

The rest of the changes look good

**Comment by JunqiZhang0** [comment]
@rujutashinde Yes, now it's ready for review

**Review by rujutashinde — CHANGES_REQUESTED** [review]
Could you add a negative test, where the error message is being caught?

**Comment by JunqiZhang0** [comment]
> Could you add a negative test, where the error message is being caught?

Done

**Review by shay6 — APPROVED** [review]

File: tests/assets/show_test_var.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+var: hellohello
```
[the remark this answers is not in the document]
Comment (JunqiZhang0) [reply tests/assets/show_test_var.yml:1]: Yup I can fix it

File: tests/assets/show_vars_data.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+---
+name: Test -vars-data for show command
+description: "Test -vars-data for show command"
+provision:
+  - name: "{{ var + '-hello' }}"
+    groups: local
+    ip_address: 127.0.0.1
+    ansible_params:
+      ansible_connection: local
+    labels: ['hello1','hello2']
```
[the remark this answers is not in the document]
Comment (JunqiZhang0) [reply tests/assets/show_vars_data.yml:10]: fixed

**Review by rujutashinde — APPROVED** [review]
LGTM

**Pull request merged**